src/RootScene.js

```diff
@@ -72,6 +74,17 @@ export default class RootScene extends Component {
     return route.transition || this.props.defaultTransition;
   }
 
+  handleRouteChange = () => {
+    const navigator = this.getNavigator();
+    // Stack diff can be either 0 or 2, but never 1.
+    if (
+      (navigator && this.routeStack && navigator.state.routeStack.length) ===
+      (this.routeStack.length + 1)
+    ) {
+      this.props.dispatch(actionCreators.pop());
+    }
```

Comment:
I'd set `this.routeStack` to null here to make it clear that that variable is only meant to be used here.